frontend/src/app/testQueue/useLocalQueue.ts

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useState } from "react";
+import { QueueItemData } from "./TestQueue";
+
+export function useLocalQueue(
+  remoteQueue: QueueItemData[],
+  refreshTime: number
+) {
+  const [localQueue, setLocalQueue] = useState<QueueItemData[]>();
+  const [toRemove, setToRemove] = useState(new Set<string>());
+
+  useEffect(() => {
+    let timeout: ReturnType<typeof setTimeout>;
+    if (!remoteQueue) {
+      return;
+    }
+    // Immediately set local queue if nonexistent
+    if (localQueue === undefined) {
+      setLocalQueue(remoteQueue);
```

Comment:
This is confirming my biggest concern about the PR. I really don't love the idea of creating a local copy of the data that could get out of sync in weird ways just for the purpose of creating a nice animation.

Do you know of any other ways to accomplish this? I don't know how it can be done without maintaining the info somehow since the components will otherwise just immediately unmount.

Reply:
Yeah i was wondering how other folks animate their lists. We could force a re-poll at the end of the animation I suppose, rather than caching it here?

Reply:
The ask on this one is to animate cards away when data from the backend comes back without one of the items a user is currently looking at. So I'm not sure how to anticipate which ones to animate away without having a local copy that stays around until the end of the animation.

Reply:
Oh right, the ones we want to animate are _already_ gone in the current copy. Ugh. I guess animations don't really pair well with the whole reactive model unless you make a copy just for animating. Well like I said, I think the timeout takes care of the local copy being consistent, but if an update occurs during an animation I have no idea how that looks. It shouldn't be _broken_ though.